habitat_baselines/rl/ddppo/policy/running_mean_and_var.py

```diff
@@ -32,15 +48,21 @@ def forward(self, x: Tensor) -> Tensor:
                 x.transpose(1, 0).contiguous().view(x.size(1), -1)
             )
             new_mean = x_channels_first.mean(-1, keepdim=True)
-            new_count = torch.full_like(self._count, n)
+            new_count = torch.full_like(self._count, n, dtype=torch.float32)
 
             if distrib.is_initialized():
-                distrib.all_reduce(new_mean)
                 distrib.all_reduce(new_count)
                 new_mean /= distrib.get_world_size()
 
+                new_mean = new_mean.float()
+                distrib.all_reduce(new_mean)
```

Comment:
Since you've updated PyTorch, all these distrib.all_reduce can be made async super easily, and they should for the sake of speed. Just make async_op = True and use the returned async op handles.

Reply:
If we are trying to eke out a little bit more efficiency here, going to 1 all_reduce call is always better (even with the async ones), so I did that.